arches_lingo/src/arches_lingo/components/generic/resource-instance-relationships/ResourceInstanceRelationshipsEditor.vue

```diff
@@ -26,27 +74,151 @@ const value = computed({
+                    displayValue: $gettext("Add a new %{graphName}", {
+                        graphName: graphInfo.name,
+                    }),
+                    graphId: graphInfo.graphid,
+                }),
+            );
+        } else {
+            options.value = [...options.value, ...references];
+        }
+
+        resourceResultsPage.value = resourceData.current_page;
+        resourceResultsTotalCount.value = resourceData.total_results;
+    } catch (error) {
+        toast.add({
+            severity: ERROR,
+            life: DEFAULT_ERROR_TOAST_LIFE,
+            summary: $gettext("Failed to fetch data."),
+            detail: error instanceof Error ? error.message : undefined,
+        });
+
+        options.value = [];
+        resourceResultsPage.value = 1;
+        resourceResultsTotalCount.value = 0;
+    } finally {
+        isLoading.value = false;
+        isLoadingAdditionalResults.value = false;
+    }
+}
+
+async function onLazyLoadResources(event: VirtualScrollerLazyEvent) {
+    console.log("ohai", event);
+    if (
+        event.last >= resourceResultsPage.value * RESULTS_PER_PAGE &&
+        event.last <= resourceResultsTotalCount.value
+    ) {
+        isLoadingAdditionalResults.value = true;
+        const page = resourceResultsPage.value + 1;
+        fetchData(page);
+    }
+}
+
+function createNewResource(graphId: string) {
+    showNewResource.value = true;
+    console.log(graphId);
+}
 </script>
 <template>
     <MultiSelect
         v-model="primeVuePickerVal"
         :show-toggle-all="!!options?.length"
         :options
-        option-label="display_value"
+        option-label="displayValue"
         option-value="resourceId"
+        class="resource-instance-relationships-selector"
+        :virtual-scroller-options="{
+            itemSize: ITEM_HEIGHT,
+            lazy: true,
+            onLazyLoad: onLazyLoadResources,
+            scrollHeight: computedResourceResultsHeight,
+            style: {
+                minHeight: computedResourceResultsHeight,
+                maxHeight: computedResourceResultsHeight,
+            },
+        }"
         :pt="{
             emptyMessage: { style: { fontFamily: 'sans-serif' } },
             option: { style: { fontFamily: 'sans-serif' } },
+            overlay: { style: { fontFamily: 'sans-serif' } },
         }"
+        :loading="isLoading && !isLoadingAdditionalResults"
         :placeholder="$gettext('Select Resources')"
         :aria-labelledby="props.ptAriaLabeledBy"
-    />
+    >
+        <template #footer>
+            <div
+                v-for="element in newElements"
+                :key="`new-${element.graphId}`"
+            >
+                <Button
+                    class="relationship-footer-btn"
+                    :label="element.displayValue"
+                    severity="secondary"
+                    variant="text"
+                    @click="createNewResource(element.graphId)"
```

Comment:
I think this would cause an invoke on initial render?

Reply:
That doesn't appear to be the case, but I've enclosed it in an anonymous function for safety.